packages/slate/src/changes/at-range.js

```diff
@@ -1366,7 +1368,20 @@ Changes.wrapInlineAtRange = (change, range, inline, options = {}) => {
   const startIndex = startBlock.nodes.indexOf(startChild)
   const endIndex = endBlock.nodes.indexOf(endChild)
 
-  if (startBlock == endBlock) {
+  if (startInline && startInline == endInline) {
+    const text = startInline.text.substring(startOffset, endOffset)
+    inline = inline.set('nodes', List([Text.create(text)]))
```

Comment:
It may result in lost marks since you are taking the string and reconstructing a `Text`, right ?

Maybe on `Node`, we should have a method `Node.getSubText(startOffset: number, endOffset: number): Text` ?

Reply:
Thanks. I'll add a test with overlapping marks and see if I can somehow construct the element with marks in place.